src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -0,0 +1,167 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class GraphPlannerPlanAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.PLANNER_PLAN_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'title', 'createdDateTime', 'owner'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const endpointPlanner: string = `${this.resource}/v1.0/planner/plans`;
+    if (args.options.ownerGroupId) {
+      this.getGroupsWithFilter(`?$filter=ID eq '${args.options.ownerGroupId}'`)
+        .then((groups: any[]): void => {
+          if (groups && groups.length > 0) {
+            if (groups.length > 1) {
+              logger.log(`More than one groups found with id ${args.options.ownerGroupId}`);
+              cb();
+            }
+            else {
+              const requestOptions: any = {
```

Comment:
Let's refactor this code so that we're not repeating creating the plan multiple times. We should start with resolving the group ID and once we have it, we'd proceed with creating the plan.